test/Microsoft.AspNet.Razor.Test/TestFiles/CodeGenerator/CS/Output/BasicTagHelpers.CustomAttributeCodeGenerator.cs

```diff
@@ -36,20 +36,20 @@ public BasicTagHelpers()
                 __tagHelperExecutionContext.Add(__PTagHelper);
                 __tagHelperExecutionContext.Output = __tagHelperRunner.RunAsync(__tagHelperExecutionContext).Result;
                 WriteLiteral(__tagHelperExecutionContext.Output.GenerateStartTag());
-                WriteLiteral(__tagHelperExecutionContext.Output.GeneratePreContent());
+                Write(__tagHelperExecutionContext.Output.GeneratePreContent());
                 if (__tagHelperExecutionContext.Output.ContentSet)
                 {
-                    WriteLiteral(__tagHelperExecutionContext.Output.GenerateContent());
+                    Write(__tagHelperExecutionContext.Output.GenerateContent());
                 }
                 else if (__tagHelperExecutionContext.ChildContentRetrieved)
                 {
-                    WriteLiteral(__tagHelperExecutionContext.GetChildContentAsync().Result);
+                    Write(__tagHelperExecutionContext.GetChildContentAsync().Result);
                 }
                 else
                 {
                     __tagHelperExecutionContext.ExecuteChildContentAsync().Wait();
                 }
-                WriteLiteral(__tagHelperExecutionContext.Output.GeneratePostContent());
+                Write(__tagHelperExecutionContext.Output.GeneratePostContent());
                 WriteLiteral(__tagHelperExecutionContext.Output.GenerateEndTag());
```

Comment:
not necessarily a part of this PR: suggest we create `WriteLiteral(TagHelperExecutionContext)` and `WriteLiteralTo(TagHelperExecutionContext)` overloads in `RazorPage`.  no reason to generate the boilerplate code from `WriteLiteral(__tagHelperExecutionContext.Output.GenerateStartTag());` through this line.  possible to include the `RunAsync()` call in the methods but that doesn't fit well into a "WriteSomething()` method.